PointSDK.iOS/ApiDefinition.cs

```diff
@@ -34,93 +36,133 @@ interface IBDPTempoTrackingDelegate
         void DidStopRequiringUserInterventionForBluetooth();
 
-        // @optional -(void)didStartRequiringUserInterventionForLocationServicesAuthorizationStatus:(CLAuthorizationStatus)authorizationStatus;
+        // @optional -(void)didStartRequiringUserInterventionForLocationServicesAuthorizationStatus:(CLAuthorizationStatus)authorizationStatus __attribute__((deprecated("First deprecated in 15.4.0 - Feature replaced by `-[BDPBluedotServiceDelegate locationAuthorizationDidChangeFromPreviousStatus:toNewStatus:]`")));
         [Abstract]
         [Export("didStartRequiringUserInterventionForLocationServicesAuthorizationStatus:")]
         void DidStartRequiringUserInterventionForLocationServicesAuthorizationStatus(CLAuthorizationStatus authorizationStatus);
 
-        // @optional -(void)didStopRequiringUserInterventionForLocationServicesAuthorizationStatus:(CLAuthorizationStatus)authorizationStatus;
+        // @optional -(void)didStopRequiringUserInterventionForLocationServicesAuthorizationStatus:(CLAuthorizationStatus)authorizationStatus __attribute__((deprecated("First deprecated in 15.4.0 - Feature replaced by `-[BDPBluedotServiceDelegate locationAuthorizationDidChangeFromPreviousStatus:toNewStatus:]`")));
         [Abstract]
         [Export("didStopRequiringUserInterventionForLocationServicesAuthorizationStatus:")]
         void DidStopRequiringUserInterventionForLocationServicesAuthorizationStatus(CLAuthorizationStatus authorizationStatus);
 
-        // @optional -(void)didStartRequiringUserInterventionForPowerMode;
+        // @optional -(void)didStartRequiringUserInterventionForPowerMode __attribute__((deprecated("First deprecated in 15.4.0 - Feature replaced by `-[BDPBluedotServiceDelegate lowPowerModeDidChange:]`")));
         [Abstract]
         [Export("didStartRequiringUserInterventionForPowerMode")]
         void DidStartRequiringUserInterventionForPowerMode();
 
-        // @optional -(void)didStopRequiringUserInterventionForPowerMode;
+        // @optional -(void)didStopRequiringUserInterventionForPowerMode __attribute__((deprecated("First deprecated in 15.4.0 - Feature replaced by `-[BDPBluedotServiceDelegate lowPowerModeDidChange:]`")));
         [Abstract]
         [Export("didStopRequiringUserInterventionForPowerMode")]
         void DidStopRequiringUserInterventionForPowerMode();
     }
 
+    // @protocol BDPBluedotServiceDelegate <NSObject>
+    [Protocol, Model]
+    [BaseType(typeof(NSObject))]
+    interface IBDPBluedotServiceDelegate
+    {
+        // @optional -(void)locationAuthorizationDidChangeFromPreviousStatus:(CLAuthorizationStatus)previousAuthorizationStatus toNewStatus:(CLAuthorizationStatus)newAuthorizationStatus;
+        [Export("locationAuthorizationDidChangeFromPreviousStatus:toNewStatus:")]
+        void LocationAuthorizationDidChangeFromPreviousStatus(CLAuthorizationStatus previousAuthorizationStatus, CLAuthorizationStatus newAuthorizationStatus);
```

Comment:
I think we need to add a ticket to our backlog to copy API documentation from the iOS SDK project. At the moment it's quite disjointed and a developer needs to refer to the our current API documentation to find out more about methods. We should have something like this:
```csharp
        /// <summary>
        /// This method can be implemented to keep track of changes in Location Manager’s CLAuthorizationStatus.
        /// This method is called whenever location authorization status changes.
        /// You may wish to communicate to your user when location authorization status changes. For example, you may wish to ask for “Always” or “When in Use” authorization from within your app.
        /// </summary>
        /// <param name="previousAuthorizationStatus">Previous authorization status</param>
        /// <param name="newAuthorizationStatus">Current authorization status</param>
        [Export("locationAuthorizationDidChangeFromPreviousStatus:toNewStatus:")]
        void LocationAuthorizationDidChangeFromPreviousStatus(CLAuthorizationStatus previousAuthorizationStatus, CLAuthorizationStatus newAuthorizationStatus);
```

Reply:
Ticket created in backlog